lib/web/fetch/index.js

```diff
@@ -2145,7 +2145,14 @@ async function httpNetworkFetch (
             const keys = Object.keys(rawHeaders)
             for (let i = 0; i < keys.length; ++i) {
               // The header names are already in lowercase.
-              headersList.append(keys[i], rawHeaders[keys[i]], true)
+              const values = rawHeaders[keys[i]]
+              if (Array.isArray(values)) {
+                for (const value of values) {
+                  headersList.append(keys[i], value, true)
+                }
+              } else {
+                headersList.append(keys[i], values, true)
+              }
```

Comment:
I think this is better.
```suggestion
              const key = keys[i]
              const value = rawHeaders[key]
              if (key === 'set-cookie') {
                for (let j = 0; j < value.length; ++j) {
                  headersList.append(key, value[j], true)
                }
              } else {
                headersList.append(key, value, true)
              }
```